src/migration/02112022/migrate-groups-dynamodb-02112022.ts

```diff
@@ -0,0 +1,144 @@
+import { EntityManager } from "@typedorm/core";
+import { GroupModel, GroupModelLatest } from "infrastructure/group-store";
+import { LoggerService } from "logger/logger";
+
+export const migrateGroupsGeneratedBy = async ({
+  entityManager,
+  loggerService,
+}: {
+  entityManager: EntityManager;
+  loggerService: LoggerService;
+}) => {
+  const latestsGroupsItems = await entityManager.find(
+    GroupModelLatest,
+    {},
+    {
+      queryIndex: "GSI1",
+    }
+  );
+  for (const group of latestsGroupsItems.items) {
+    const migratedGroup = mapToGroupGenerator(group, loggerService);
+    await entityManager.create(migratedGroup, {
+      overwriteIfExists: true,
+    });
+
+    const allGroups = await entityManager.find(GroupModel, {
+      name: group.name,
+    });
+
+    for (const groupGenerated of allGroups.items) {
+      const migratedGroupGenerated = mapToGroupGenerator(
+        groupGenerated,
+        loggerService
+      );
+      await entityManager.create(migratedGroupGenerated, {
+        overwriteIfExists: true,
+      });
+    }
+  }
+
+  return entityManager;
+};
+
+const mapToGroupGenerator = (
+  group: GroupModel | GroupModelLatest,
+  loggerService: LoggerService
+): GroupModel | GroupModelLatest => {
+  if (group instanceof GroupModel) {
+    loggerService?.info(`Migrating Group ${group.name} as a GroupModel`);
+  } else {
+    loggerService?.info(`Migrating Group ${group.name} as a GroupModelLatest`);
+  }
+  for (const mapping of Object.entries(groupGeneratorMapping)) {
+    const regexp = new RegExp(mapping[0]);
+    if (regexp.test(group.name)) {
+      loggerService?.info(
+        `Group ${group.name} has been generated with ${mapping[1]}`
+      );
+      group.generatedBy = mapping[1];
+      return group;
+    }
+  }
+  // add a fake group generator for every groups that have been saved to the DB but don't match any regex
```

Comment:
throw if it's the case? It will be better to handle all group-generators by hand.
For the groups that does not have a group generator anymore we could have a "_deleted" flag something like that